sdks/python/apache_beam/utils/exception.py

```diff
@@ -0,0 +1,32 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+def raise_exception(tp, value, tb):
+  """
+  Raise Exception from exc_info.
+  Derived from `future.utils.raise_`
+  """
+  if value is not None and isinstance(tp, Exception):
+    raise TypeError("instance exception may not have a separate value")
+  if value is not None:
+    exc = tp(value)
```

Comment:
I'm not sure this fixes the problem described in the bug report. What if value is not None but `tp` takes more than one argument? Can you add a unit test to verify that this case is handled correctly?